app/src/main/java/com/duckduckgo/app/featureusage/FeatureSegmentsManagerImpl.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.featureusage
+
+import com.duckduckgo.app.featureusage.db.FeatureSegmentsDataStore
+import com.duckduckgo.app.pixels.AppPixelName
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.APP_TP_ENABLED
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.BOOKMARKS_IMPORTED
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.EMAIL_PROTECTION_SET
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.FAVOURITE_SET
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.FIRE_BUTTON_USED
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.FIVE_SEARCHES_MADE
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.LOGIN_SAVED
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.SET_AS_DEFAULT
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.TEN_SEARCHES_MADE
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentType.TWO_SEARCHES_MADE
+import com.duckduckgo.app.statistics.api.featureusage.FeatureSegmentsManager
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+class FeatureSegmentManagerImpl @Inject constructor(
+    private val featureSegmentsDataStore: FeatureSegmentsDataStore,
+    private val pixel: Pixel,
+) : FeatureSegmentsManager {
```

Comment:
This class is implementing the interface from the statistics module but it doesn't require anything specific from the app module?

Reply:
As I mentioned before, I will move it to `statistics` module